flexget/components/ftp/sftp.py

```diff
@@ -1,130 +1,22 @@
-)
-
-# retry configuration constants
-CONNECT_TRIES = 3
-RETRY_INTERVAL = 15
-RETRY_STEP = 5
-SOCKET_TIMEOUT = 15
-
-# make separate path instances for local vs remote path styles
-localpath = os.path
-remotepath = posixpath  # pysftp uses POSIX style paths
-
-try:
-    import pysftp
-
-    logging.getLogger("paramiko").setLevel(logging.ERROR)
-except ImportError:
-    pysftp = None
-
-
-def sftp_connect(conf):
-    """
-    Helper function to connect to an sftp server
-    """
-    sftp = None
-    tries = CONNECT_TRIES
-    retry_interval = RETRY_INTERVAL
-
-    while not sftp:
-        try:
-            sftp = pysftp.Connection(
-                host=conf.host,
-                username=conf.username,
-                private_key=conf.private_key,
-                password=conf.password,
-                port=conf.port,
-                private_key_pass=conf.private_key_pass,
-            )
-            sftp.timeout = SOCKET_TIMEOUT
-            logger.verbose('Connected to {}', conf.host)
-        except Exception as e:
-            if not tries:
-                raise e
-            else:
-                logger.debug('Caught exception: {}', e)
-                logger.warning(
-                    'Failed to connect to {}; waiting {} seconds before retrying.',
-                    conf.host,
-                    retry_interval,
-                )
-                time.sleep(retry_interval)
-                tries -= 1
-                retry_interval += RETRY_STEP
-
-    return sftp
-
-
-def sftp_from_config(config):
-    """
-    Creates an SFTP connection from a Flexget config object
-    """
-    host = config['host']
-    port = config['port']
-    username = config['username']
-    password = config['password']
-    private_key = config['private_key']
-    private_key_pass = config['private_key_pass']
-
-    conn_conf = ConnectionConfig(host, port, username, password, private_key, private_key_pass)
-
-    try:
-        sftp = sftp_connect(conn_conf)
-    except Exception as e:
-        raise plugin.PluginError('Failed to connect to %s (%s)' % (host, e))
-
-    return sftp
-
-
-def sftp_prefix(config):
-    """
-    Generate SFTP URL prefix
-    """
-    login_str = ''
-    port_str = ''
-
-    if config['username'] and config['password']:
-        login_str = '%s:%s@' % (config['username'], config['password'])
-    elif config['username']:
-        login_str = '%s@' % config['username']
 
-    if config['port'] and config['port'] != 22:
-        port_str = ':%d' % config['port']
-
-    return 'sftp://%s%s%s/' % (login_str, config['host'], port_str)
-
-
-def dependency_check():
-    """
-    Check if pysftp module is present
-    """
-    if not pysftp:
-        raise plugin.DependencyError(
-            issued_by='sftp',
-            missing='pysftp',
-            message='sftp plugin requires the pysftp Python module.',
-        )
+SftpConfig = namedtuple(
```

Comment:
I love namedtuples. Good call here!

Reply:
It  was already there, but I authored the original plugin 5 years ago so thanks I guess. 😂